pybamm/models/submodels/thermal/current_collector/base_current_collector_thermal.py

```diff
@@ -0,0 +1,220 @@
+#
+# Base class for thermal effects which accounts for current collectors
+#
+import pybamm
+
+
+class BaseModel(pybamm.BaseSubModel):
+    """Base class for thermal effects which accounts for current collectors.
+    Note: this can only be used for lumped, 1+1D or 2+1D thermal models in which
+    the temperature is independent of the through-cell direction x.
+
+
+    Parameters
+    ----------
+    param : parameter class
+        The parameters to use for this submodel
+
+
+    **Extends:** :class:`pybamm.BaseSubModel`
+    """
+
+    def __init__(self, param):
+        super().__init__(param)
+
+    def _get_standard_fundamental_variables(self, T_av):
+        param = self.param
+
+        T_cn = T_av
+        T_n = pybamm.PrimaryBroadcast(T_av, ["negative electrode"])
+        T_s = pybamm.PrimaryBroadcast(T_av, ["separator"])
+        T_p = pybamm.PrimaryBroadcast(T_av, ["positive electrode"])
+        T_cp = T_av
+
+        # Note: T is the temperature defined only in the negative electrode,
+        # separator and positive electrode. This is the quanitity used in other
+        # submodels (e.g. electrolyte)
+        T = pybamm.Concatenation(T_n, T_s, T_p)
+
+        T_volume_av = self._yz_average(T_av)
```

Comment:
This will only be valid for the lumped thermal model then? (this is the most relevant one). But, do you think that if we wanted to do the full thermal model, we would need two separate current collector?

Reply:
all of the Nplus1D models are in the limit where T doesn't depend on x (this is partly why I did a different base class for the current collector stuff, since its fundamental variable is really the x-averaged temperature. I think to do the full model (i.e. T a function of x, y and z) we would need to put in a mesh for each current collector so you can apply BCs at the edges of the current collector

Reply:
Ah okay so all current collector thermal models are x-lumped models. Whereas "lumped-model" here means "x-y-z lumped"

Reply:
I think with regards to inheritance structure would be somthing like: 

thermal/
  base_thermal.py
  x-lumped/
    base_x_lumped.py
    1D_x_lumped.py
    (current collector models)
  x-full/
    1D_full.py 
    
?

Reply:
yeah that structure would work. yeah I was taking lumped to be equivalent to volume averaged I guess. I think the structure you've put here in nice though. happy to leave merging this for now, and merge after restructuring

Reply:
I am making these changes on this branch now. Basic structure is implemented but still need to change elsewhere to pass tests.